teos/src/responder.rs

```diff
@@ -203,21 +210,15 @@ impl Responder {
         breach: Breach,
         user_id: UserId,
     ) -> ConfirmationStatus {
-        // Do not add already added trackers. This can only happen if handle_breach is called twice with the same data, which can only happen
-        // if Watcher::block_connected is interrupted during execution and called back during bootstrap.
-        if let Some(tracker) = self.trackers.lock().unwrap().get(&uuid) {
-            return tracker.status;
-        }
-
         let mut carrier = self.carrier.lock().unwrap();
         let tx_index = self.tx_index.lock().unwrap();
 
         // Check whether the transaction is in mempool or part of our internal txindex. Send it to our node otherwise.
-        let status = if carrier.in_mempool(&breach.penalty_tx.txid()) {
+        let status = if let Some(block_hash) = tx_index.get(&breach.penalty_tx.txid()) {
+            ConfirmationStatus::ConfirmedIn(tx_index.get_height(block_hash).unwrap() as u32)
+        } else if carrier.in_mempool(&breach.penalty_tx.txid()) {
```

Comment:
I don't know why I think there was a reasoning behind checking the mempool first, but I cannot really recall why.

This seems a better approach though, given we don't interact with `bitcoind` if not necessary. However, I feel I'm missing something :/